ui/src/AppCharting/chart-data-extractor.ts

```diff
@@ -44,33 +44,25 @@ const FUNCTION_MAP: Record<ChartType, ChartFactoryCallback> = {
   [BY_SIZE]: getChartDataBySize
 };
 
-
 export function chartDataFactory(type: ChartType): ChartFactoryCallback {
   const fn = FUNCTION_MAP[type];
   if (!fn) throw new Error(`Chart data for ${type} is not currently supported`);
-
-  return async (
-    benchmarkMap: Benchmarks,
-    timestamps: TimestampViewModel[],
-    opts: FormConfiguration
-  ): Promise<Data> => {
-    const data = await fn(benchmarkMap, timestamps, opts);
-    return mutating_assignNames(data);
-  };
+  return fn;
```

Comment:
It made sense to return the function directly, so we could add an equity unit test on the factory function.